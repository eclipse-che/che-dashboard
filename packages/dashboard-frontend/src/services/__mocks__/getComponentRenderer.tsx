/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

export default function <T extends Array<unknown>>(
  getComponent: (..._args: T) => React.ReactElement,
): {
  createSnapshot: (...args: T) => renderer.ReactTestRenderer;
  renderComponent: (...args: T) => {
    reRenderComponent: (..._args: T) => void;
  };
} {
  return {
    createSnapshot: (...args: T) => {
      return renderer.create(getComponent(...args));
    },
    renderComponent: (...args: T) => {
      const res = render(getComponent(...args));
      return {
        reRenderComponent: (..._args: T) => {
          res.rerender(getComponent(..._args));
        },
      };
    },
  };
}
