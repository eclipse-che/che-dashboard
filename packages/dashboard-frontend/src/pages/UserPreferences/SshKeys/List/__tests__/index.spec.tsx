/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { sshKey1, sshKey2 } from '@/pages/UserPreferences/SshKeys/__tests__/stub';
import getComponentRenderer, { screen, within } from '@/services/__mocks__/getComponentRenderer';

import { Props, SshKeysList } from '..';

jest.mock('../Entry');

const { createSnapshot, renderComponent } = getComponentRenderer(getComponent);

describe('SSH keys list', () => {
  const props: Props = {
    sshKeys: [sshKey1, sshKey2],
    onDeleteSshKeys: jest.fn(),
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('snapshot', () => {
    const snapshot = createSnapshot(props);
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  it('should display the SSH keys', () => {
    renderComponent(props);

    const entries = screen.getAllByTestId('ssh-keys-list-entry');

    expect(entries).toHaveLength(2);

    expect(within(entries[0]).getByText(sshKey1.name)).toBeInTheDocument();
    expect(within(entries[1]).getByText(sshKey2.name)).toBeInTheDocument();
  });

  it('should delete the SSH key', () => {
    renderComponent(props);

    const entries = screen.getAllByTestId('ssh-keys-list-entry');

    const deleteButton = within(entries[0]).getByRole('button', { name: 'Delete' });
    expect(deleteButton).toBeInTheDocument();

    deleteButton.click();

    expect(props.onDeleteSshKeys).toHaveBeenCalledWith([sshKey1]);
  });
});

function getComponent(props: Props): React.ReactElement {
  return <SshKeysList {...props} />;
}
