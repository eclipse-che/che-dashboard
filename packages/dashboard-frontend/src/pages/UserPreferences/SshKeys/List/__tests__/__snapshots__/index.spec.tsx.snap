// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSH keys list snapshot 1`] = `
<section
  className="pf-c-page__main-section"
>
  <div
    data-testid="ssh-keys-list-entry"
  >
    <span>
      key-name-1
    </span>
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
  <div
    data-testid="ssh-keys-list-entry"
  >
    <span>
      key-name-2
    </span>
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</section>
`;
